### PR TITLE
Fix nullpointer exception in AbstractDockerMojo

### PR DESCRIPTION
**Pull request by detached**
When there is no configuration section in settings.xml a nullpointer occurs.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -163,9 +169,27 @@
...
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.assertj</groupId>
+      <artifactId>assertj-core</artifactId>
+      <version>3.0.0</version>
+      <scope>test</scope>
+    </dependency>
+      <dependency>
+          <groupId>org.springframework</groupId>
```
Comment (davidxia) [inline pom.xml]: 2 space indent

**Comment by davidxia** [comment]
@detached Thanks for this. `mvn compile` works but `mvn test` gives me

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project docker-maven-plugin: Compilation failure: Compilation failure:
[ERROR] /Users/dxia/src/docker-maven-plugin/src/test/java/com/spotify/docker/UtilsTest.java:[51,9] cannot access java.util.Optional
[ERROR] class file for java.util.Optional not found
[ERROR] /Users/dxia/src/docker-maven-plugin/src/test/java/com/spotify/docker/UtilsTest.java:[57,9] cannot access java.time.ZonedDateTime
[ERROR] class file for java.time.ZonedDateTime not found
[ERROR] /Users/dxia/src/docker-maven-plugin/src/test/java/com/spotify/docker/UtilsTest.java:[63,9] cannot access java.time.LocalDateTime
[ERROR] class file for java.time.LocalDateTime not found
[ERROR] /Users/dxia/src/docker-maven-plugin/src/test/java/com/spotify/docker/UtilsTest.java:[69,9] cannot access java.time.LocalTime
[ERROR] class file for java.time.LocalTime not found
[ERROR] /Users/dxia/src/docker-maven-plugin/src/test/java/com/spotify/docker/UtilsTest.java:[75,9] cannot access java.time.LocalDate
[ERROR] class file for java.time.LocalDate not found
```

We also use 2 space indents.

**Comment by detached** [comment]
It is a compatibility issue with Java7. 
The build only works with Java8  https://travis-ci.org/detached/docker-maven-plugin/builds
I will correct this.

**Comment by davidxia** [comment]
@detached Hey, sorry I completely forgot about this. Can you rebase from master?

File: README.md [hunk 2]
```diff
@@ -125,6 +125,9 @@ credentials in your Maven's global `settings.xml` file as part of the `<servers>
         <id>docker-hub</id>
         <username>foo</username>
         <password>secret-password</password>
+        <configuration>
+            <email>foo@foo.bar</email>
```
Comment (davidxia) [inline README.md]: 2 space indent

File: pom.xml [hunk 3]
```diff
@@ -148,6 +148,12 @@
       <artifactId>maven-plugin-testing-harness</artifactId>
       <version>3.1.0</version>
       <scope>test</scope>
+        <exclusions>
```
Comment (davidxia) [inline pom.xml]: 2 space indent

File: pom.xml [hunk 4]
```diff
@@ -163,10 +169,27 @@
...
     </dependency>
-
+    <dependency>
+      <groupId>org.assertj</groupId>
+      <artifactId>assertj-core</artifactId>
+      <version>2.0.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.springframework</groupId>
```
Comment (davidxia) [inline pom.xml]: spacing

File: pom.xml [hunk 5]
```diff
@@ -163,10 +169,27 @@
...
+        <artifactId>spring-test</artifactId>
+        <version>4.1.6.RELEASE</version>
+        <scope>test</scope>
+    </dependency>
+    <dependency>
+        <groupId>junit</groupId>
```
Comment (davidxia) [inline pom.xml]: spacing

File: pom.xml (hunk 4 above)
Comment (davidxia) [inline pom.xml]: Why do we need this dependency?

**Comment by detached** [comment]
Sorry I forget about this, too. I will close the merge request, fix the spacing stuff, rebase and squash and reopen again.

**Pull request closed without merging**